### ISSUE TITLE
`RPC failed with status = "Unavailable: Socket closed"`

### ISSUE THREAD

**Issue by ibeltagy** [pytorch/xla#2361]
## 🐛 Bug

My setting was working fine last week, then today started throwing the following error then crashing. This happens during the first gradient update. 

```
2020-07-19 18:52:43.783429: W    2250 tensorflow/core/distributed_runtime/rpc/grpc_remote_master.cc:160] RPC failed
 with status = "Unavailable: Socket closed" and grpc_error_string = "{"created":"@1595184763.783222230",
"description":"Error received from peer ipv4:10.164.133.234:8470","file":"external/com_github_grpc_grpc/src/core/lib/surface/call.cc", 
"file_line":1056,"grpc_message":"Socket closed","grpc_status":14}", maybe retrying the RPC
```

Here's the debug information: [debug.tar.gz](https://github.com/pytorch/xla/files/4944432/debug.tar.gz)

**Comment by dlibenzi** [comment]
Hi @ibeltagy , is this repeatable?
Can you try to restart the TPU node, and fetch latest nightly?
The clients logs simple report a vanishing session, which might be the service side crashing, or some network issue.

**Comment by ibeltagy** [comment]
Yes, I tried restarting the tpu and creating a new nightly one, both didn't help.

**Comment by dlibenzi** [comment]
Any chance of providing a repro?
Is this Colab/Kaggle or a real GCE VM?
This is probably caused by a service side issue, whose logs we cannot access, so having a repro could help trigger from our environment and see what's inside the service side lggs.

**Comment by ibeltagy** [comment]
This is a GCE VM. Here's how to reproduce 
```
# Install requirements 
pip install transformers==2.11.0
pip install pytorch-lightning==0.8.5

# Train
import torch
from torch.utils.data import DataLoader, Dataset
from transformers import AutoModel
import pytorch_lightning as pl

class CoolDataset(Dataset):
  def __len__(self):
      return 128 * 128

  def __getitem__(self, idx):
      return torch.tensor([1, 2, 3, 4,] * 128), torch.tensor([1, 1, 1, 1] * 128)

class CoolSystem(pl.LightningModule):

    def __init__(self):
        super().__init__()
        self.model = AutoModel.from_pretrained('bert-base-uncased')

    def forward(self, x, y):
        return self.model(x, attention_mask=None)

    def training_step(self, batch, batch_idx):
        x, y = batch
        y_hat = self(x, y)
        loss = y_hat[0].sum() #  F.cross_entropy(y_hat, y)
        tensorboard_logs = {'train_loss': loss}
        return {'loss': loss, 'log': tensorboard_logs}

    def validation_step(self, batch, batch_idx):
        x, y = batch
        y_hat = self(x, y)
        val_loss = y_hat[0].sum() #  F.cross_entropy(y_hat, y)
        return {'val_loss': val_loss}

    def validation_epoch_end(self, outputs):
        avg_loss = torch.stack([x['val_loss'] for x in outputs]).mean()
        tensorboard_logs = {'val_loss': avg_loss}
        return {'val_loss': avg_loss, 'log': tensorboard_logs}

    def configure_optimizers(self):
        return torch.optim.Adam(self.parameters(), lr=0.001)

    def train_dataloader(self):
        loader = DataLoader(CoolDataset(), batch_size=1, num_workers=0)
        return loader

    def val_dataloader(self):
        loader = DataLoader(CoolDataset(), batch_size=1, num_workers=0)
        return loader

if __name__ == '__main__':
    model = CoolSystem()
    trainer = pl.Trainer(num_tpu_cores=1, progress_bar_refresh_rate=1, max_epochs=10, num_sanity_val_steps=0)
    trainer.fit(model)
```

**Comment by JackCaoG** [comment]
Yup, It is a server side crash. seems to be related to the scatter.
`Log line format: [IWEF]mmdd hh:mm:ss.uuuuuu threadid file:line] msg
F0720 18:35:45.799373     901 scatter_emitter.cc:1993] Check failed: fusion_parameter->user_count() == 1 (2 vs. 1)`

I will take a closer look.

**Comment by dlibenzi** [comment]
Thanks @JackCaoG !
Just file an XLA bug.

**Comment by JackCaoG** [comment]
@ibeltagy I filed a bug internally, will update in this thread when we have a better idea.

**Comment by ibeltagy** [comment]
thanks.

**Comment by JackCaoG** [comment]
@ibeltagy TPU side fix merged, if you could use tomorrow's nightly and select the latest runtime this runtime error should be gone.

**Comment by ibeltagy** [comment]
ok, will wait until tomorrow. Is it enough to restart the tpu or do I need to delete it and create a new one?

**Comment by JackCaoG** [comment]
Restart the TPU should be enough. I usually use the command `curl -X POST http://$TPU_IP:8475/requestversion/pytorch-devyyyymmdd` to manually fetch the runtime at `yyyymmdd` which is more reliable and don't need to wait for TPU to restart.

**Comment by ibeltagy** [comment]
Thanks.

**Comment by ibeltagy** [comment]
It is working. Thanks, @JackCaoG.

**Comment by steveepreston** [comment]
Can someone say, which trying to log via `[IWEF]mmdd hh:mm:ss.uuuuuu threadid file:line]`?
how can we mute that completely?
